leaper.py

```diff
@@ -125,35 +127,48 @@ def check_source_submission(self, src_project, src_package, src_rev, target_proj
         src_srcinfo = self.get_sourceinfo(src_project, src_package, src_rev)
         package = target_package
 
+        origin = None
+
         if src_srcinfo is None:
             # source package does not exist?
             # handle here to avoid crashing on the next line
             self.logger.warn("Could not get source info for %s/%s@%s" % (src_project, src_package, src_rev))
             return False
 
         if self.ibs and target_project.startswith('SUSE:SLE'):
+            if package in self.lookup_sp3:
+                origin = self.lookup_sp3[package]
+
+            origin_same = True
+            if origin:
+                origin_same = True if origin == 'FORK' else src_project.startswith(origin)
+                self.logger.info("expected origin is '%s' (%s)", origin,
+                                 "unchanged" if origin_same else "changed")
+
             # True or None (open request) are acceptable for SLE.
-            if self._check_factory(target_package, src_srcinfo, 'openSUSE.org:openSUSE:Factory') is not False:
-                self.logger.info('found package in openSUSE.org:openSUSE:Factory')
-                return True
+            self.source_in_factory = self._check_factory(package, src_srcinfo, 'openSUSE.org:openSUSE:Factory')
+            if self.source_in_factory is None:
+                self.pending_factory_submission = True
```

Comment:
I did not set this since the same style of message would make sense for the rest of the checks so I kept it consistent. The way the messages are built is a bit separated from the rest and perhaps a function with a project as a parameter would be better.

Obviously, this can be done for just Factory if desired.